File: src/test/java/test/UpdateDepartmentTest.java
package test;
import cn.authing.sdk.java.client.ManagementClient;
import cn.authing.sdk.java.dto.*;
import cn.authing.sdk.java.model.ManagementClientOptions;
import cn.authing.sdk.java.util.JsonUtils;

import java.util.Collections;



public class UpdateDepartmentTest {

    private static final String ACCESS_KEY_ID = "AUTHING_ACCESS_KEY_ID";
    private static final String ACCESS_KEY_SECRET = "AUTHING_ACCESS_KEY_SECRET";

    public static void main(String[] args) throws Throwable {
        ManagementClientOptions clientOptions = new ManagementClientOptions();
        clientOptions.setAccessKeyId(ACCESS_KEY_ID);
        clientOptions.setAccessKeySecret(ACCESS_KEY_SECRET);
        ManagementClient client = new ManagementClient(clientOptions);

        UpdateDepartmentReqDto request = new UpdateDepartmentReqDto();
        request.setDepartmentId("departmentId_8023");
        request.setOrganizationCode("organizationCode_9338");
        request.setLeaderUserIds(Collections.singletonList("String_2681"));
        request.setDescription("description_5390");
        request.setCode("code_7002");
        request.setI18n(new cn.authing.sdk.java.dto.DepartmentI18nDto());
        request.setName("name_1648");
        request.setDepartmentIdType(cn.authing.sdk.java.dto.UpdateDepartmentReqDto.DepartmentIdType.DEPARTMENT_ID);
        request.setParentDepartmentId("parentDepartmentId_7650");
        request.setCustomData(Collections.emptyList());

        DepartmentSingleRespDto response = client.updateDepartment(request);
        System.out.println(JsonUtils.serialize(response));
    }

}

File: src/test/go/test/UpdateDepartmentTest.go
package main

import (
"authing-go-sdk/client"
"authing-go-sdk/dto"
"fmt"
)

func main() {
    options := client.ManagementClientOptions{
        AccessKeyId:     "AUTHING_USERPOOL_ID",
        AccessKeySecret: "AUTHING_USERPOOL_SECRET",
    }
    var err error
    client, err := client.NewClient(&options)
    if err != nil {
        panic(err)
    }
    request := dto.UpdateDepartmentReqDto{
        DepartmentId: "departmentId_8086",
        OrganizationCode: "organizationCode_7102",
        LeaderUserIds: nil,
        Description: "description_5761",
        Code: "code_1093",
        I18n: nil,
        Name: "name_7858",
        DepartmentIdType: ,
        ParentDepartmentId: "parentDepartmentId_7472",
        CustomData: nil,
    }
    response := client.UpdateDepartment(&request)
    fmt.Println(response)
}

